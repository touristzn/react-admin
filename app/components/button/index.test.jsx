import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Button from './index'

let wrapper

describe('<Button />', () => {
  const props = {
    label: 'edit',
    // type和size都可自定义值，style.less中加入样式即可
    // 目前type有: primary / secondary / disabled
    type: 'secondary',
    onClick: jest.fn(),
    // 不传为默认大小
    size: 'small'
  }

  beforeEach(() => {
    wrapper = shallow(
      <Button {...props} />
    )
  })

  it('renders the text of the label', () => {
    expect(wrapper.find('button').text()).toBe(props.label)
  })

  it('property type and size are rendered correctly', () => {
    expect(wrapper.hasClass('btn-secondary')).toEqual(true)
    expect(wrapper.hasClass('btn-small')).toEqual(true)
  })

  it('click the Button should trigger the function', () => {
    wrapper.simulate('click')
    expect(props.onClick).toHaveBeenCalled()
  })

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})