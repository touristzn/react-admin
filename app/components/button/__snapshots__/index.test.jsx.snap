// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<button
  className="btn btn-secondary btn-small"
  disabled={false}
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  type="button"
>
  edit
</button>
`;
